### PR TITLE
SPOI-10330  HDFS S3 Sync Application

### PR DESCRIPTION
**Pull request by deepak-narkhede**
Changes includes:
    1) HDFS to S3 Application.
    2) Local mode Test application.
    3) README file.

Tested on local and cluster mode with some sample output.
Also more sample output is going on.

### CODE REVIEW

**Comment by deepak-narkhede** [comment]
@yogidevendra . Please review.

File: hdfs-to-s3-sync/src/main/resources/META-INF/properties.xml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    <name>dt.operator.HDFSInputModule.prop.files</name>
+    <value>/tmp/deepak/hdfsToS3</value>
+  </property>
+
+  <!--
+   S3 Output Module properties
+   !-->
+  <property>
+    <name>dt.operator.S3OutputModule.prop.accessKey</name>
+    <value>accessKey</value>
```
Comment (yogidevendra) [inline hdfs-to-s3-sync/src/main/resources/META-INF/properties.xml]: Keep default values consistent with https://github.com/DataTorrent/app-templates/blob/master/s3-to-hdfs-sync/src/main/resources/META-INF/properties.xml

File: hdfs-to-s3-sync/src/main/java/com/datatorrent/apps/Application.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+     */
+    FSInputModule inputModule = dag.addModule("HDFSInputModule", new FSInputModule());
+    S3OutputModule outputModule = dag.addModule("S3OutputModule", new S3OutputModule());
+
+    /*
+     * Create a stream for Metadata blocks from HDFS to S3 output modules.
+     * Note: DAG locality is set to CONTAINER_LOCAL for performance benefits by
+     * avoiding any serialization/deserialization of objects.
+     */
+    dag.addStream("FileMetaData", inputModule.filesMetadataOutput, outputModule.filesMetadataInput);
```
Comment (yogidevendra) [inline hdfs-to-s3-sync/src/main/java/com/datatorrent/apps/Application.java]: Better to use THREAD_LOCAL instead. Refer https://github.com/DataTorrent/app-templates/blob/master/s3-to-hdfs-sync/src/main/java/com/datatorrent/apps/Application.java

Comment (deepak-narkhede) [inline hdfs-to-s3-sync/src/main/java/com/datatorrent/apps/Application.java]: To make it THREAD_LOCAL, the S3OutputModule operators especially S3InitiateFileUploader must be thread local too. Hence the validation fails for the data stream for blockdata.
Hence keeping it CONTAINER_LOCAL.

File: hdfs-to-s3-sync/src/main/resources/META-INF/properties.xml (hunk 1 above)
Comment (deepak-narkhede) [inline hdfs-to-s3-sync/src/main/resources/META-INF/properties.xml]: Done.

**Pull request merged**